### PR TITLE
Add customer_managed_key functionality

### LINKED ISSUE #15: [FEAT] customer_managed_key
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add the ability to specify a customer_managed_key when creating the storage account.

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Potential Terraform Configuration

```hcl
module "storage-account" {
  
  customer_managed_key = {
    key_vault_key_id          = local.keyvault_key_name
    user_assigned_identity_id = azurerm_user_assigned_identity.identity.id
  }
}
```

### References

[customer_managed_key](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#customer_managed_key)

### PR DESCRIPTION
**Pull request by blongv**
Fixes #15  .

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Changes proposed in this pull request

- Add outputs
- Add customer_managed_key optional block.
- Add infrastructure_encryption_enabled variable.
- Add lifecycle ignore_changes tags for environments where tags are enforced by central policies.

@claranet/fr-azure-reviewers

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -26,6 +26,16 @@ output "storage_account_network_rules" {
   )
 }
 
+output "storage_account_primary_blob_endpoint" {
```
Comment (Shr3ps) [inline outputs.tf:29]: This is not needed for your added feature "customer_managed_key" and this endpoint is already available through the `storage_account_properties` output

File: variables.tf [hunk 2]
```diff
@@ -141,6 +141,22 @@ variable "identity_ids" {
 
 # Data protection
 
+variable "customer_managed_key" {
```
Comment (Shr3ps) [inline variables.tf:144]: please use tfdocs to update the README/fix the CI

**Review by Shr3ps — CHANGES_REQUESTED** [review]

**Review by Shr3ps — APPROVED** [review]

File: r-storage-account.tf [hunk 3]
```diff
@@ -43,6 +45,15 @@ resource "azurerm_storage_account" "storage" {
     }
   }
 
+  dynamic "customer_managed_key" {
+    for_each = var.customer_managed_key != null ? ["enabled"] : []
```
Comment (zfiel) [inline r-storage-account.tf:49]:
```suggestion
    for_each = var.customer_managed_key[*] 
```

File: r-storage-account.tf [hunk 4]
```diff
@@ -43,6 +45,15 @@ resource "azurerm_storage_account" "storage" {
...
+    content {
+      key_vault_key_id          = try(var.customer_managed_key.key_vault_key_id, null)
+      managed_hsm_key_id        = try(var.customer_managed_key.managed_hsm_key_id, null)
```
Comment (zfiel) [inline r-storage-account.tf:52]:
```suggestion
      key_vault_key_id          = var.customer_managed_key.key_vault_key_id
      managed_hsm_key_id  = var.customer_managed_key.managed_hsm_key_id
```

**Review by zfiel — APPROVED** [review]

**Pull request merged**